src/test/16_ActionMultiCall.t.sol

```diff
@@ -0,0 +1,372 @@
+
+import {ActionMultiCall} from "../actions/MultiCall.sol";
+import "../actions/utils/ActionData.sol";
+
+import {TrustedProtocolMock} from "../mockups/TrustedProtocolMock.sol";
+import {LendingPool, DebtToken, ERC20} from "../../lib/arcadia-lending/src/LendingPool.sol";
+import {Tranche} from "../../lib/arcadia-lending/src/Tranche.sol";
+
+contract VaultTestExtension is Vault {
+    function setAllowed(address who, bool allow) public {
+        allowed[who] = allow;
+    }
+
+    function setTrustedProtocol(address trustedProtocol_) public {
+        trustedProtocol = trustedProtocol_;
+    }
+
+    function setIsTrustedProtocolSet(bool set) public {
+        isTrustedProtocolSet = set;
+    }
+}
+
+contract ActionMultiCallTest is DeployArcadiaVaults {
+    using stdStorage for StdStorage;
+
+    ActionMultiCall public action;
+    MultiActionMock public multiActionMock;
+    LendingPool public pool;
+    DebtToken public debt;
+    Tranche public tranche;
+
+    VaultTestExtension public proxy_;
+    TrustedProtocolMock public trustedProtocol;
+
+    function setUp() public {
+        action = new ActionMultiCall();
+        deal(address(eth), address(action), 1000 * 10 ** 20, false);
+
+        vm.startPrank(creatorAddress);
+        vault = new VaultTestExtension();
+        factory.setNewVaultInfo(address(mainRegistry), address(vault), Constants.upgradeProof1To2);
+        factory.confirmNewVaultInfo();
+        vm.stopPrank();
+
+        vm.startPrank(vaultOwner);
+        proxyAddr = factory.createVault(12345678, 0);
+        proxy_ = VaultTestExtension(proxyAddr);
+        vm.stopPrank();
+
+        depositERC20InVault(eth, 1000 * 10 ** 18, vaultOwner);
+
+        vm.startPrank(creatorAddress);
+        mainRegistry.setAllowedAction(address(action), true);
+
+        pool = new LendingPool(ERC20(address(dai)), creatorAddress, address(factory));
+        pool.setLiquidator(address(liquidator));
+        pool.setVaultVersion(1, true);
+        debt = DebtToken(address(pool));
+        tranche = new Tranche(address(pool), "Senior", "SR");
+        pool.addTranche(address(tranche), 50);
+
+        trustedProtocol = new TrustedProtocolMock();
+        vm.stopPrank();
+    }
+
+    function testSuccess_executeAction() public {
+        actionAssetsData memory assetData = actionAssetsData({
+            assets: new address[](1),
+            assetIds: new uint256[](0),
+            assetAmounts: new uint256[](1),
+            assetTypes: new uint256[](1),
+            preActionBalances: new uint256[](0)
+        });
+
+        assetData.assets[0] = address(eth);
+        assetData.assetTypes[0] = 0;
+
+        address[] memory to = new address[](1);
+        bytes[] memory data = new bytes[](1);
+        to[0] = address(this);
+        data[0] = abi.encodeWithSignature("returnFive()");
+
+        bytes memory callData = abi.encode(assetData, assetData, to, data);
+
+        action.executeAction(address(0), callData);
```

Comment:
Does test indeed return 5?

Reply:
not possible to check... however if function isn't found it will revert, so it does find it (also seen in traces)

Reply:
I'llchange it to update a var and check whether it's updated